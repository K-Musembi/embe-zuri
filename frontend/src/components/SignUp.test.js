import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import SignUp from './SignUp';

// Mock fetch API
global.fetch = jest.fn(() =>
  Promise.resolve({
    ok: true,
    json: () => Promise.resolve({ message: 'Success' })
  })
);
global.alert = jest.fn();

describe('SignUp Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <SignUp onSignUp={() => {}} />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
