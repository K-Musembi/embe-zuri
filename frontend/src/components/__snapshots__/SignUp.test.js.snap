// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Component renders correctly 1`] = `
<div
  className="signup-login-container"
>
  <h2>
    Create Account
  </h2>
  <form
    onSubmit={[Function]}
  >
    <label>
      Username
      <input
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </label>
    <label>
      Email
      <input
        onChange={[Function]}
        type="email"
        value=""
      />
    </label>
    <label>
      password
      <input
        onChange={[Function]}
        type="password"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Sign Up
    </button>
  </form>
</div>
`;
